test/addons-napi/test_error/test_error.cc

```diff
@@ -15,9 +15,64 @@ napi_value checkError(napi_env env, napi_callback_info info) {
   return result;
 }
 
+napi_value throwExistingError(napi_env env, napi_callback_info info) {
+  napi_value message;
+  napi_value error;
+  NAPI_CALL(env, napi_create_string_utf8(env, "existing error", -1, &message));
+  NAPI_CALL(env, napi_create_error(env, message, &error));
+  NAPI_CALL(env, napi_throw(env, error));
+  return nullptr;
```

Comment:
The `napi_create_error()` and `napi_throw_error()` and similar APIs are intended to be used by native add-on modules to return and/or throw errors to JavaScript. So, it would be up to the native add-on code to supply meaningful values for the `code` property. But, N-API could encourage conformance to that convention by providing APIs such as `napi_create_error_with_code()`, `napi_throw_error_with_code()`. I'll open a new issue to track that.